src/sfapi_client/_async/storage.py

```diff
@@ -0,0 +1,231 @@
+        >>>     res = await globus.transfer(
+                        "globus-transfer-uuid"
+                )
+
+        :param int timeout: time to wait for the transfer to complete, defaults to sys.maxsize
+        :return GlobusStart: Gives the file status for the transfer
+        """
+        return await self._wait_until_complete(timeout)
+
+    async def _fetch_state(self):
+        r = await self.globus.client.get(f"storage/globus/transfer/{self.transfer_id}")
+        json_response = r.json()
+        json_response["transfer_id"] = self.transfer_id
+        json_response["globus"] = self.globus
+        transfer = AsyncGlobusTransfer.model_validate(json_response)
+        return transfer
+
+
+class AsyncGlobus(StorageBase):
+    client: Optional["AsyncClient"]  # noqa: F821
+    source_machine: Optional[Union[Machine, str, None]]
+    target_machine: Optional[Union[Machine, str, None]]
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    @check_auth
+    async def start_transfer(
+        self,
+        source_dir: Union[str, Path, AsyncRemotePath],
+        target_dir: Union[str, Path, AsyncRemotePath],
+        label: Optional[str] = None,
+    ) -> AsyncGlobusTransfer:
+        """Start a Globus transfer throught the SuperFacility API
+
+        - Must select the Globus option when creating the SuperFacility key
+
+        ```python
+        >>> from sfapi_client import AsyncClient
+        >>> async with AsyncClient(client_id, client_secret) as client:
+        >>>     globus = client.storage.globus(Machine.dtns, Machine.dtns)
+        >>>     res = await globus.start_transfer(
+                        "/pscratch/sd/u/user/globus",
+                        "/global/cfs/cdirs/m0000/globus"
+                    )
+        ```
+
+        :param str source_dir: Path to file or directory on the source to transfer
+        :param str target_dir: Path to directory on the target to transfer files to
+        :param Optional[str] label: Label for the transfer,
+        defaults to None and the API will create a label for the transfer
+        :return AsyncGlobusTransfer
+        """
+
+        if None in [source_dir, target_dir]:
+            # Check that all parametes are not none
+            raise ValueError("source_dir, and target_dir cannot be None")
+
+        # Make machine names match those in the API endpoint
+        source_name = (
+            "dtn"
+            if self.source_machine in [Machine.dtns, Machine.dtn01]
+            else self.source_machine
+        )
+        target_name = (
+            "dtn"
+            if self.target_machine in [Machine.dtns, Machine.dtn01]
+            else self.target_machine
+        )
+
+        body = {
+            "source_uuid": source_name,
+            "target_uuid": target_name,
+            "source_dir": source_dir,
+            "target_dir": target_dir,
+            "label": label,
+        }
```

Comment:
You shoud be able to use the `BodyStartGlobusTransferStorageGlobusTransferPost` pydantic model, the nice thing about this is it will keep us in sync with any changes made on the server side.